### DISCUSSION TITLE
Easy way to extract user-defined parameters?

### DISCUSSION THREAD

**Discussion by RobinM92** [Open-Systems-Pharmacology/Forum#1854] [Questions & Problems]
Dear community,

When reporting PBPK modeling work, we of course like to report on all parameters that were changed from the default value. In PKSim there is a covenient tab within the simulation called 'user-defined' where all these parameters are listed. Is there a straightforward way (e.g. using the OSP-suite-R or esqlabsR package) to extract this list of parameters, including name and value in R?

Thanks for your insights!
Robin

**Comment by PavelBal** [comment]
Unfortunately, I am not aware of such functionality to extract user-defined parameters from a pkml.

In the [`Scenario`-based workflow](https://esqlabs.github.io/esqlabsR/articles/esqlabsR-design-scenarios.html) in `{esqlabsR}`, you can get all custom parameter that are applied to the loaded pkml with `Scenario$finalCustomParams`. An approach would be to always export a simulation with default parameters to pkml and only to change parameters using the `Scenarios` workflow in esqlabsR. Then you can easily see which parameters were modified in the workflow.
